test/spec/EnrollWindowsHello_spec.js

```diff
@@ -0,0 +1,180 @@
+/*jshint maxparams:17, es5:false */
+define([
+  'vendor/lib/q',
+  'underscore',
+  'jquery',
+  '@okta/okta-auth-js/jquery',
+  'util/Util',
+  'helpers/mocks/Util',
+  'helpers/dom/EnrollWindowsHelloForm',
+  'helpers/dom/Beacon',
+  'helpers/util/Expect',
+  'sandbox',
+  'vendor/webauthn',
+  'LoginRouter',
+  'helpers/xhr/MFA_ENROLL_allFactors',
+  'helpers/xhr/MFA_ENROLL_Webauthn'
+],
+function (Q, _, $, OktaAuth, LoginUtil, Util, Form, Beacon, Expect, $sandbox, webauthn, Router, responseMfaEnrollAll,
+          responseMfaEnrollWebauthn) {
+
+  var itp = Expect.itp;
+  var tick = Expect.tick;
+
+  Expect.describe('EnrollWindowsHello', function () {
+
+    function setup() {
+      var setNextResponse = Util.mockAjax([responseMfaEnrollAll, responseMfaEnrollWebauthn]);
+      var baseUrl = 'https://foo.com';
+      var authClient = new OktaAuth({url: baseUrl, transformErrorXHR: LoginUtil.transformErrorXHR});
+      var router = new Router({
+        el: $sandbox,
+        baseUrl: baseUrl,
+        authClient: authClient,
+        globalSuccessFn: function () {}
+      });
+      Util.mockRouterNavigate(router);
+      return tick()
+      .then(function () {
+        router.refreshAuthState('dummy-token');
+        return tick();
+      })
+      .then(function () {
+        router.enrollWindowsHello();
+        return tick(null);
+      })
+      .then(function () {
+        return {
+          router: router,
+          beacon: new Beacon($sandbox),
+          form: new Form($sandbox),
+          ac: authClient,
+          setNextResponse: setNextResponse
+        };
+      });
+    }
+
+    function emulateNotWindows() {
+      spyOn(webauthn, 'isAvailable').and.returnValue(false);
+      spyOn(webauthn, 'makeCredential');
+      spyOn(webauthn, 'getAssertion');
+
+      return tick();
+    }
+
+    function emulateWindows() {
+      spyOn(webauthn, 'isAvailable').and.returnValue(true);
+
+      spyOn(webauthn, 'makeCredential').and.callFake(function () {
+        var deferred = $.Deferred();
```

Comment:
Not sure using $.Deferred is a good indicator of the Promise spec - if you want something closer, use Q. Or just use the Promise API since that's what we're actually seeing in the code.